src/sourcegraph.d.ts

```diff
@@ -785,6 +785,37 @@ declare module 'sourcegraph' {
         ): Unsubscribable
     }
 
+    /**
+     * A query transformer alters a user's search query before executing a search.
+     *
+     * Query transformers allow extensions to define new search query operators and syntax, for example,
+     * by matching strings in a query (e.g. `go.imports:`) and replacing them with a regular expression or string.
+     */
+    export interface QueryTransformer {
+        /**
+         * Transforms a search query into another, valid query. If there are no transformations to be made
+         * the original query is returned.
+         *
+         * @param query A search query provided by a client.
```

Comment:
just "A search query" is fine, no need to say "provided by a client" (not sure what that adds)